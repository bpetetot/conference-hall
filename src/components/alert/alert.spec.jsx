/* eslint-env jest */
import React, { Fragment } from 'react'
import { shallow } from 'enzyme'
import Button from 'components/button'
import Alert from './index'

describe('components/alert', () => {
  it('should render with a title only', () => {
    const wrapper = shallow(<Alert title="boo" />)
    expect(wrapper).toMatchSnapshot()
  })
  it('should render with type info icon', () => {
    const wrapper = shallow(<Alert title="boo" type="info" />)
    expect(wrapper.find('i.fa-info-circle').exists()).toBeTruthy()
  })
  it('should render with type info error', () => {
    const wrapper = shallow(<Alert title="boo" type="error" />)
    expect(wrapper.find('i.fa-exclamation-circle').exists()).toBeTruthy()
  })
  it('should render with type info warning', () => {
    const wrapper = shallow(<Alert title="boo" type="warning" />)
    expect(wrapper.find('i.fa-exclamation-triangle').exists()).toBeTruthy()
  })
  it('should render with type info sucess', () => {
    const wrapper = shallow(<Alert title="boo" type="success" />)
    expect(wrapper.find('i.fa-check-circle').exists()).toBeTruthy()
  })
  it('should render with action buttons', () => {
    const comp = () => (
      <Fragment>
        <Button primary>
          I confirm my venue
        </Button>
        <Button primary error>
          I cancel my venue
        </Button>
      </Fragment>
    )
    const wrapper = shallow(<Alert
      title="boo"
      actionButtons={comp}
    />)
    expect(wrapper).toMatchSnapshot()
  })
})
