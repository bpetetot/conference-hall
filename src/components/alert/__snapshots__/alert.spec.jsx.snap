// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/alert should render with a title only 1`] = `
<div
  className="alert"
>
  <span
    className="alert-icon alert-icon-info"
  >
    <i
      className="fa fa-info-circle fa-2x"
    />
  </span>
  <span
    className="alert-title"
  >
    boo
  </span>
  <div
    className="actions"
  />
</div>
`;

exports[`components/alert should render with action buttons 1`] = `
<div
  className="alert"
>
  <span
    className="alert-icon alert-icon-info"
  >
    <i
      className="fa fa-info-circle fa-2x"
    />
  </span>
  <span
    className="alert-title"
  >
    boo
  </span>
  <div
    className="actions"
  >
    <Button
      accent={false}
      block={false}
      error={false}
      loading={false}
      primary={true}
      secondary={false}
      simple={false}
      size="normal"
      tertiary={false}
    >
      I confirm my venue
    </Button>
    <Button
      accent={false}
      block={false}
      error={true}
      loading={false}
      primary={true}
      secondary={false}
      simple={false}
      size="normal"
      tertiary={false}
    >
      I cancel my venue
    </Button>
  </div>
</div>
`;
